pingcap/ticdc
# owner: Modified the update strategy of  gcSafePoint

### What problem does this PR solve? 
TiKV GC safe point will be blocked indefinitely due to TiCDC changefeed checkpoint stagnation.

### What is changed and how it works?
Set a threshold Y: Regardless of the status of the changefeed, TiCDC is allowed to block TiKV GC safe time for Y hours at most.

The threshold Y is set in the config/config.go file as a variable named gc-TTL and the default value is 84000 seconds (24 hours).

### Check List 

Tests 
 - Unit test

 - Manual test (add detailed scripts or steps below)
step1: set  `defaultServerConfig.gcTTL = 20` in /pkg/config.go 

step2: add ` log.Warn("try to update gcSafePoint", zap.Time("gcSafePoint", oracle.GetTimeFromTS(gcSafePoint)))`  below line 802 in cdc/owner.go to print log info

step3: In a shell window type `tiup playground` to deploy a tidb cluster.

step4: Move to ticdc directory, and type `./bin/cdc server` to to start the cdc component。

step5: Type `./cdc cli changefeed create --pd=http://127.0.0.1:2379 --sink-uri="mysql://root@127.0.0.1:4000/" --changefeed-id="ttl-test-10"` to create a changefeed.

step6: pause above changefeed, you would see gcSafePoint stop update:
![image](https://user-images.githubusercontent.com/20351731/116217877-93a85d00-a77c-11eb-856a-b7970f143d8e.png)

step7: gcSafePoint update again after stop 20s:
![image](https://user-images.githubusercontent.com/20351731/116218490-26e19280-a77d-11eb-9417-217e9e32e75d.png)



Code changes

 - Has exported function/method change

Side effects
 - Possible performance regression

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Release note


- Modified the update strategy of gcSafePoint.  Fix the problem that TiKV GC safe point is blocked indefinitely due to TiCDC changefeed checkpoint stagnation.
